src/app/query-explainer/components/reproduced-result-section/reproduced-result-section.component.ts

```diff
@@ -10,20 +10,59 @@ import { DataRow, DataSchemaField } from "src/app/interface/dataset.interface";
 })
 export class ReproducedResultSectionComponent {
     @Input({ required: true }) public output: QueryExplainerOutputType;
+    @Input({ required: true }) public dataFormat: keyof typeof QueryExplainerDataFormat;
 
     public tableSource(output: QueryExplainerOutputType): DataRow[] {
-        const result = output.data.map((dataItem) => {
-            const arr = dataItem.map((value, index) => ({ [this.columnNames(output.schema)[index]]: value }));
+        const columnNames: string[] = output.schema.fields.map((item) => item.name);
+        switch (this.dataFormat) {
+            case "JsonAoS": {
+                return this.parseDataFromJsonAoSFormat(output.data, columnNames);
+            }
+            case "JsonAoA": {
+                return this.parseDataFromJsonAoAFormat(output.data, columnNames);
+            }
+            case "JsonSoA": {
+                return this.parseDataFromJsonSoAFormat(output.data, columnNames);
+            }
+            /* istanbul ignore next */
+            default:
+                throw new Error("Unable to parse date");
```

Comment:
As discussed - UI should modify `input.dataFormat` to request data in a format that suits it best.

Reply:
Removed this code